amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/filters/LocationFilterListManager.java

```diff
@@ -112,14 +118,28 @@ public FilterList getFilterList() {
      * @return Country ids with children
      */
     protected List<Long> getCountriesWithChildrenIds() {
-        String query = "SELECT acvl.id FROM amp_category_value_location acvl "
-                + "WHERE acvl.parent_location IS NULL "
-                + "AND acvl.id IN (SELECT DISTINCT parent_location FROM amp_category_value_location)";
         
-        Collection<BigInteger> countryCollection = PersistenceManager.getSession().createSQLQuery(query).list();
+        Session session = PersistenceManager.getSession();
+        
+        String queryString = "SELECT loc FROM " + AmpCategoryValueLocations.class.getName()
+                + " loc WHERE loc.parentLocation IS NULL "
+                + " AND (loc.deleted != true)"
+                + " AND (loc.id IN (SELECT DISTINCT parentLocation FROM "
+                + AmpCategoryValueLocations.class.getName() + "))";
+        
+        Query qry = session.createQuery(queryString);
+        qry.setCacheable(true);
+        Collection<AmpCategoryValueLocations> countryCollection = qry.list();
+    
+        AmpApplicationSettings appSettings = EndpointUtils.getAppSettings();
+        final boolean showAllCountries = appSettings == null ? false : appSettings.getShowAllCountries();
+    
+        final String defaultIso = FeaturesUtil.getGlobalSettingValue(GlobalSettingsConstants.DEFAULT_COUNTRY);
```

Comment:
Lets change `defaultIso` to `defaultCountryIso` so its more clear for anyone reading the code